infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -131,10 +187,21 @@ def restore_database(backup_fpath):
     )
     run_shell_command(create_db_cmd, env_vars=pg_env_vars)
 
+    # find the most recent backup or verify the specify backup exists
+    print("Getting backup file from S3")
+    session = create_s3_session()
+    if backup_obj_name is None:
+        backup_obj_name = get_most_recent_backup_obj_name(session)
```

Comment:
Rather than mutate `backup_obj_name` could you create a new variable for the name of the object that is actually used? Maybe `object_name_of_database_backup_to_restore_from` 🤣.

This would do it. No need for an `if` statement :D 

```python
object_name_of_database_backup_to_restore_from: str = None or get_most_recent_backup_obj_name(session)
```